File: ruby_solution/line_parser.rb
require_relative './item'

class LineParser

  def self.parse_line(line)
    name = line[line.index(' ')..line.index(' at ')].sub("imported ", "").strip!
    quantity = line[0..line.index(' ')].to_i
    price = line[line.index('at ')+3..line.length].to_f
    is_imported = line.include? "imported" 
    Item.new(name, quantity, price, is_imported)
  end

end

File: java_solution/LineParser.java
public class LineParser {

  public static Item parseLine(String line) {

    String name = line.substring(line.indexOf(" "), line.indexOf(" at ")).replace(" imported", "").trim();
    int quantity = Integer.parseInt(line.substring(0, line.indexOf(" ")));
    double price = Double.parseDouble(line.substring(line.indexOf("at ")+3, line.length()));
    boolean isImported = line.indexOf("imported") >= 0;
    return new Item(name, quantity, price, isImported);
  }

}